apps/settings/js/panels/about/update_check.js

```diff
@@ -33,11 +33,35 @@ define(function(require) {
 
       this._loadLastUpdated();
 
+      this._displayNotifyCheckbox();
+
       this._elements.checkUpdateNow.addEventListener('click',
         this._checkForUpdates.bind(this));
     },
 
     /**
+     * Shows and hides the "notify me" checkbox depending on whether auto
+     * updates are enabled.
+     *
+     * @access private
+     * @memberOf UpdateCheck.prototype
+     */
+    _displayNotifyCheckbox: function uc__displayNotifyCheckBox() {
+      var key = 'addons.auto_update';
+      var request = this._settings.createLock().get(key);
```

Comment:
settings lock supports promise interface AFAIK, please use that so we have both then and catch screnarios (console.error on catch please).